src/main/java/org/cirdles/topsoil/app/utils/SVGSaver.java

```diff
@@ -37,35 +43,69 @@
     public static final String SVG_DOCTYPE_PUBLIC = "-//W3C//DTD SVG 1.1//EN";
     private static final String SVG_DOCTYPE_SYSTEM = "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd";
 
+    
+    /**
+     * Saves the vector image in Topsoil with the .svg file extension
+     * 
+     * @param svgDocument   Document to be saved
+     */
     public void save(Document svgDocument) {
+       
+        // If there is no file specified, then there is nothing to do.
+        // Otherwise, write the file with the SVG format.
+        generateSaveUI().ifPresent(outStream -> {
+            writeSVGToOutputStream(svgDocument, outStream);
+        });
+
+    }
+    
+    /**
+     * Create a save dialog window and prompt user for filepath
+     * 
+     * @return Optional<File>   the filepath
+     */
+    public Optional<OutputStream> generateSaveUI() {              
         // get the path for the new SVG file
         FileChooser fileChooser = new FileChooser();
         fileChooser.setTitle("Export to SVG");
         fileChooser.setInitialDirectory(new File(System.getProperty("user.home")));
-        fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("SVG Image", "*.svg"));
-        File svgFile = fileChooser.showSaveDialog(null);
-
-        // can't convert if the file isn't specified
-        if (svgFile == null) {
-            return;
+        fileChooser.getExtensionFilters().setAll(
+                new FileChooser.ExtensionFilter("All Files", "*"),
+                new FileChooser.ExtensionFilter("SVG Image", "*.svg")               
+        );
+        Optional<OutputStream> outStream = Optional.empty();
+        try {
+            outStream = Optional.ofNullable(Files.newOutputStream(fileChooser.showSaveDialog(null).toPath()));
```

Comment:
You might want to introduce another method here and test it. You have a bug in this line.